### DISCUSSION TITLE
Server-side frameworks: Spring, Quarkus and Micronaut

### DISCUSSION THREAD

**Discussion by skin27** [jhipster/generator-jhipster#24580] [General]
Jhipster is a great generator for learning and trying new frameworks and libraries. On the frontend/client-side, the user gets the option to choose between Angular, React and Vue.

On the backend/server-side, there is not such an option. By default, Spring boot is used. Based on the history of Jhipster, Spring has always been the default framework, while others were not available. There are of course blueprints for other server-side frameworks like Quarkus and Micronaut. These blueprints are not on the same level as Spring, though. For example, the Quarkus blueprint doesn't have the option to generate Gateway/Microservices, though it's the field of Microservices where Quarkus shines (and Micronaut as well). Also, the current Quarkus is a bit behind (still uses Quarkus 2.x instead of 3.x) and doesn't support Java 21. 

In recent years, Quarkus and Micronaut gained a lot of traction. It's almost the default choice for microservices, but based on what I hear and read in the field, becoming the de facto choice when starting a greenfield application. There are many reasons for this, like efficient usage of resources (startup/memory/CPU) and smooth developer experience.

As Jhipster is exactly used to generate greenfield applications, and for microservices and trying out new technologies, it would make sense to offer more server-side options by default. And if only one server-side would be used, should that not be Quarkus in 2024?

**Comment by Tcharl** [comment]
Hi, you also have other blueprints like node, ....

All these blueprint are contribution friendly, and analytics speaks: spring is far behind in term of traction and user base, so for now spring is preferred. To get Quarkus or Micronaut first class citizen, contribution should keep the blueprint at the level of expectation, and jhipster download analytics should prove that the switch should be done
